omod/src/test/java/org/openmrs/web/rest/controller/TagRestControllerTest.java

```diff
@@ -0,0 +1,73 @@
+package org.openmrs.web.rest.controller;
+
+import java.util.List;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Tag;
+import org.openmrs.TagConstants;
+import org.openmrs.api.TagService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+public class TagRestControllerTest extends MainResourceControllerTest {
+	
+	TagService tagService;
+	
+	public static final String TAG_RESOURCE_DATASET = "TagResourceDataSet.xml";
+	
+	@Before
+	public void setup() throws Exception {
+		tagService = Context.getService(TagService.class);
+		executeDataSet(TAG_RESOURCE_DATASET);
+	}
+	
+	@Override
+	public String getURI() {
+		return TagConstants.TAG_URI;
+	}
+	
+	@Override
+	public String getUuid() {
+		return TagConstants.TAG_UUID;
+	}
+	
+	@Override
+	public long getAllCount() {
+		return tagService.getTags("").size();
```

Comment:
You should not be passing in an empty space, you need to remove this behavior in the TagService if you implemented it this way. There should be a specific method to get all tags that takes no arguments